src/Cardano/Shell/Types.hs

```diff
@@ -53,9 +71,9 @@ initializeCardanoEnvironment = do
         , ceEkgStore    = ekgStore
         }
 
-loadCardanoConfiguration :: IO CardanoConfiguration
-loadCardanoConfiguration = do
-    pure $ CardanoConfiguration mempty mempty mempty mempty
+loadCardanoConfiguration :: CardanoConfiguration -> IO CardanoConfiguration
+loadCardanoConfiguration configuration = do
```

Comment:
This seems obsolete then unless you pass all the things that `CardanoConfiguration` requires.

Reply:
Seems like I'll need to write a lot of boilerplate  `instance Monoid` for the nested data types. Perhaps https://hackage.haskell.org/package/generic-monoid is a better option?

Reply:
You could simply derive it in the deriving clause?

Reply:
I believe Generics are required, no? https://stackoverflow.com/questions/49517453/defining-a-monoid-instance-for-a-record-type

Reply:
Still getting warning about implicit <> for Semigroup, but it compiles at least.